### PR TITLE
Docs for Unity Catalog

### PR DESCRIPTION
**Pull request by bilalaslamseattle**
### Description
Documentation how to use dbt-databricks with Unity Catalog.

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

**Review by andrefurlan-db — APPROVED** [review]

File: docs/uc.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Using dbt-databricks with Unity Catalog
+
+[Unity Catalog](https://www.databricks.com/product/unity-catalog) provides a SQL interface to manage tables, views and more across Databricks workspaces using standard SQL. dbt-databricks supports 
+
+# Benefits of Unity Catalog for analytics engineers
+
+We strongly recommend using Unity Catalog for all dbt projects on Databricks. Unity Catalog brings two important benefits to analytics engineers:
+
+1. Unity Catalog offers a three-level namespace: catalog > schema > table/view. Note that schema is a synonym for database in earlier releases of dbt-databricks. This lets you manage and isolate data and avoids polluting a single database with unrelated tables and views. For example, you can data between your finance and HR teams inside a production catalog called `prod` in two schemas, `finance` and `hr`:
```
Comment (ueshin) [inline docs/uc.md:9]: > you can data between ...

Do you mean `you can isolate date` or `you can separate data` or something like that?

File: docs/uc.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    └──────┐
+           ├───── people
+           └───── salaries
+```
+
+2. All privilege management is through standard SQL, which means you don't have to learn a new method of managing access.
+3. Unity Catalog offers [automatic column-level lineage](https://docs.databricks.com/data-governance/unity-catalog/data-lineage.html).
+
+# Default catalog
+When you run `dbt init`, the CLI will ask you for a catalog. This is the _default_ catalog. Note that dbt models can refer to tables and views in any catalog (as long as you have appropriate privileges). The default catalog is simply where dbt will create tables and views by default.
```
Comment (ueshin) [inline docs/uc.md:29]: > dbt models can refer to tables and views in any catalog

We might want to mention about `hive_metastore`. Cross-catalog queries between UC and hive metastore have limitations.

Comment (bilalaslamseattle) [reply docs/uc.md]: @ueshin PTAL

**Review by ueshin — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**